gpytorch/likelihoods/likelihood.py

```diff
@@ -141,12 +140,20 @@ def name_prefix(self):
         def name_prefix(self, val):
             self._name_prefix = val
 
-        def _draw_likelihood_samples(self, function_dist, *args, **kwargs):
-            # Hack to get the current plating structure
-            sample_shape = pyro.sample("__throwaway__", pyro.distributions.Normal(0, 1)).shape
-            if not len(sample_shape):
-                sample_shape = None
-            return super()._draw_likelihood_samples(function_dist, *args, **kwargs, sample_shape=sample_shape)
+        def _draw_likelihood_samples(self, function_dist, *args, sample_shape=None, **kwargs):
```

Comment:
Maybe `_pyro_draw_likelihood_samples`? In both PRs, I think all methods with pyro primitive calls start with `pyro` or `_pyro` so far -- might be a good standard.

Reply:
I agree about the pyro prefixing in general. However `_draw_likelihood_samples` is also used for non-pyro likelihoods as well.

Reply:
If it's used for non pyro likelihoods, should we either remove the pyro statements from the method or make pyro a dependency of the package? as is, this method is uncallable in a fresh environment with `pip install gpytorch`.

Reply:
The `_draw_likelihood_samples` that you're referring to is for the base likelihood class when Pyro is available. Otherwise it uses this version of the method:

https://github.com/cornellius-gp/gpytorch/blob/f2f442fc95047b3c432d7cbf5e908084ff0d3cb7/gpytorch/likelihoods/likelihood.py#L22